dwh/indicator_lib.py

```diff
@@ -83,14 +83,54 @@ def calc_TX_PVLS(patient_agg_obs: pandas.DataFrame, VL_code: str,
     end_date = datetime.strptime(end_date_str, '%Y-%m-%d')
   temp_df = patient_agg_obs[(patient_agg_obs['code'] == VL_code)].copy()
   # Note the above copy is used to avoid setting a new column on a slice next:
-  temp_df['sup_VL'] = (temp_df['max_value'] < failure_threshold)
+  # separate value from date
+  temp_df[['vl_max_date', 'latest_vl_value']] = temp_df.max_date_value.str.split(query_lib.DATE_VALUE_SEPARATOR,
+                                                                                       expand=True)
+  temp_df['latest_vl_value'] = temp_df.loc[:, 'latest_vl_value'].astype(float)
+
+  temp_df['sup_VL'] = (temp_df.loc[:, 'latest_vl_value'] < failure_threshold)
```

Comment:
Just to clarify my previous comment about view/copy and []/loc/assign: AFAIU this mostly matters for the left-hand-side of an assignment (or when we are reusing that slice in a different scope). For temporary values on the right hand side, since we are not writing anything (only reading) it does not matter whether it is a copy or view. So please feel free to use more readable notations like `(temp_df['latest_vl_value'] < failure_threshold)` as you had before.
```suggestion
  temp_df['sup_VL'] = (temp_df['latest_vl_value'] < failure_threshold)
```